enigma/src/main/java/cuchaz/enigma/source/DecompiledClassSource.java

```diff
@@ -22,47 +22,52 @@
 	private final ClassEntry classEntry;
 
 	private final SourceIndex obfuscatedIndex;
-	private SourceIndex remappedIndex;
+	private final SourceIndex remappedIndex;
 
-	private final Map<RenamableTokenType, Collection<Token>> highlightedTokens = new EnumMap<>(RenamableTokenType.class);
+	private final Map<RenamableTokenType, Collection<Token>> highlightedTokens;
```

Comment:
Could we extract this `Map<RenamableTokenType, Collection<Token>>` into a wrapper type? Stuff like `highlightToken()` and some of the logic to calculate the new caret pos could be moved in there too.

Reply:
Sure, that's probably a good idea.